### PR TITLE
[Survival] Serpent Sting update and Viper's Venom Talent

### PR DESCRIPTION
**Pull request by Pewtro**
Also fixes a crash caused by focus capping. 

Viper's Venom:
image

Serpent Sting: 
image

Traits/Talents box: 
image

### CODE REVIEW

File: src/Parser/Hunter/Survival/Modules/Spells/SerpentSting.js [hunk 1]
```diff
@@ -3,33 +3,193 @@ import React from 'react';
...
+    if (targetInstance === undefined) {
+      targetInstance = 1;
+    }
+    const hastedSerpentStingDuration = BASELINE_DURATION / (1 + this.statTracker.currentHastePercentage);
+    const serpentStingTarget = { targetID: event.targetID, targetInstance: targetInstance, timestamp: event.timestamp };
+    for (let i = 0; i <= this.serpentStingTargets.length - 1; i++) {
+      if (this.serpentStingTargets[i].targetID === serpentStingTarget.targetID && this.serpentStingTargets[i].targetInstance === serpentStingTarget.targetInstance) {
+        const timeRemaining = this.serpentStingTargets[i].serpentStingDuration - (event.timestamp - this.serpentStingTargets[i].timestamp);
+        if (timeRemaining < (this.serpentStingTargets[i].serpentStingDuration * PANDEMIC) && !this.hasVV) {
+          this.badRefresh++;
```
Comment (nuoHep) [inline src/Parser/Hunter/Survival/Modules/Spells/SerpentSting.js]: Shouldn't this be `timeRemaining > (hastedSerpentStingDuration * PANDEMIC)`?

Comment (Pewtro) [reply src/Parser/Hunter/Survival/Modules/Spells/SerpentSting.js]: Good catch! It should indeed. Fixed.

Comment (nuoHep) [reply src/Parser/Hunter/Survival/Modules/Spells/SerpentSting.js]: Using `hastedSerpentStingDuration` is also important imo, as that's what is used by the refreshing logic (just below).

Comment (Pewtro) [reply src/Parser/Hunter/Survival/Modules/Spells/SerpentSting.js]: Like the latest commit added right?

File: src/Parser/Hunter/Survival/Modules/Features/TraitsAndTalents.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  statistic() {
+    return (
+      <StatisticsListBox
+        title="Spells, Traits and Talents"
+        tooltip="This provides an overview of the damage contributions of various talents and traits. This isn't meant as a way to 1:1 evaluate talents, as some talents bring other strengths to the table than pure damage."
+      >
+        {this.aMurderOfCrows.active && this.aMurderOfCrows.subStatistic()}
+        {this.serpentSting.active && this.serpentSting.subStatistic()}
+        {this.vipersVenom.active && this.vipersVenom.subStatistic()}
+        {}
```
Comment (MartijnHols) [inline src/Parser/Hunter/Survival/Modules/Features/TraitsAndTalents.js]: Left these brackets here

**Pull request merged**